src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -33,6 +35,72 @@ public static class EnumTests
                 @"""MyUInt64Enum"" : " + ulong.MaxValue +
                 @"}";
 
+        private static readonly string s_jsonByteEnum =
+                @"{" +
+                @"""MyByteEnum"" : " + byte.MaxValue +
+                @"}";
+
+
+        public static IEnumerable<object[]> JsonWithInvalidOutOfBoundNumber
+        {
+            get
+            {
+                const string InvalidNumber = "18446744073709551616"; // ulong.MaxValue + 1;
+                foreach(var i in new dynamic[] { InvalidNumber, ulong.MaxValue })
+                {
+                    yield return new object[] { $"{{ \"MyEnum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyEnum\" : {i} }}" };
+                    yield return new object[] { $"{{ \"MyByteEnum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyByteEnum\" : {i} }}" };
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : {i} }}" };
+                }
+                foreach(var i in new dynamic[] { InvalidNumber })
+                {
+                    yield return new object[] { $"{{ \"MyUInt64Enum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyUInt64Enum\" : {i} }}" };
+                }
+                foreach(var i in new dynamic[] { -1, (ulong)byte.MaxValue + 2 })
+                {
+                    yield return new object[] { $"{{ \"MyByteEnum\" : \"{i}\" }}" };
+                }
+                foreach(var i in new dynamic[] { -1, (ulong)UInt32.MaxValue + 2 })
+                {
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : \"{i}\" }}" };
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> JsonWithAcceptableInvalidNumber
+        {
+            get
+            {
+                foreach(var (i,j) in new (dynamic, dynamic)[] { (-1, -1), ((ulong)byte.MaxValue + 2, (ulong)UInt32.MaxValue + 2) })
```

Comment:
This is a lot of machinery, making it difficult to follow. Why not just have an array of int64 that has the test inputs?

Reply:
Also, any reason to pick `+ 2`?  We could just pick `+ 1` and then not have to add the one below when we do the modulo.